TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/TeleOp.java

```diff
@@ -257,6 +268,14 @@ public void runViperSlidesToPosition(double position, double power) {
         }
     }
 
+    public void gamepadUpdate() {
+        previousGamepad1.copy(currentGamepad1);
+        previousGamepad2.copy(currentGamepad2);
+
+        currentGamepad1.copy(gamepad1);
```

Comment:
what's the point of having a seperate "currentGamepad1"? why can't we just use gamepad1 and gamepad2?

Reply:
If we remove it previousGamepad wouldn't work because it relies on being updated before the current is updated for the previous action. gamepad1 is updating constantly so It wouldn't work.